### PR TITLE
dev/core#3692 Remove call to legacy function

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
slight variant on [ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php](https://github.com/civicrm/civicrm-core/pull/23880/files#diff-446abc52c350f634414e671e427dde8807ebf82e5b8e8aafc53895daa9b7f385R375)

### CODE REVIEW

File: ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php [hunk 1]
```diff
@@ -372,16 +372,15 @@ public function summary() {
...
     while ($dao->fetch()) {
       $row = [];
       foreach ($this->_tableFields as $name => $dontCare) {
-        if ($name != 'activity_type_id') {
+        if ($name !== 'activity_type_id') {
           $row[$name] = $dao->$name;
         }
         else {
-          $row['activity_type'] = $activityTypes[$dao->$name] ?? NULL;
+          $row['activity_type'] = CRM_Core_PseudoConstant::getLabel('CRM_Activity_BAO_Activity', 'activity_type_id', $dao->$name);
```
Comment (demeritcowboy) [inline ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php:383]: This isn't the same. See https://github.com/civicrm/civicrm-core/pull/23983#discussion_r917453363

Comment (eileenmcnaughton) [reply ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php:383]: @demeritcowboy this seems qualitively different to the original though - I think we expect custom field options to be disabled & enabled & existing rows to be fine with that - but do we really think it's 'OK' to disable an activity type when activities of that type exist and then display them (without the type) - I think implicit in activity type being a required field is that the type should not be disable-able if in use

Comment (demeritcowboy) [reply ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php:383]: I would agree with you, but 50% of people will disagree with it.

Comment (eileenmcnaughton) [reply ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php:383]: Specifically on activity_type?

Comment (demeritcowboy) [reply ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php:383]: I'll add a note to https://lab.civicrm.org/dev/core/-/issues/19.

Comment (eileenmcnaughton) [reply ext/legacycustomsearches/CRM/Contact/Form/Search/Custom/FullText.php:383]: OK - I'll close this & let it resolve itself on gitlab before putting more on github

**Pull request closed without merging**